src/golang/cmd/server/handler/register_airflow_workflow.go

```diff
@@ -92,7 +92,7 @@ func (h *RegisterAirflowWorkflowHandler) Prepare(r *http.Request) (interface{},
 		dagSummary.Dag.Metadata.Name,
 		h.Database,
 	)
-	if err != nil {
+	if err != nil && !errors.Is(err, database.ErrNoRows()) {
```

Comment:
We were missing this critical check of the error type, since `database.ErrNoRows` is actually what we want